frontend/pages/Tools/index.tsx

```diff
@@ -22,6 +23,10 @@ type AppPageProps = WithProvidersProps<PageProps>;
 export const getServerSideProps: GetServerSideProps<AppPageProps> = async (
    context
 ) => {
+   const protocol = context.req.headers.referer?.split('://')[0] || 'https';
+   const url = `${protocol}://${context.req.headers.host}${context.resolvedUrl}`;
+   setSentryPageContext({ url });
```

Comment:
What do you think about changing setSentryPageContext to accept a GetServerSideProps context object so it can calculate the url and protocol internally without having to copy/paste that. I imagine there will be something like this on every page. Not sure if there's a way to globally add this, but that could be good too. I also think for something like resolvedurl, we might want that to be a sentry tag rather than context, since tags are searchable but context isn't.

Reply:
Oh I see you used setTag down in the actual implementation so nevermind on that part.

Reply:
> What do you think about changing setSentryPageContext to accept a GetServerSideProps context object so it can calculate the url and protocol internally

Yea that seems reasonable, would definitely simplify calling it as the number of pages grows.

Reply:
I agree. FYI, Next v12 has a [middleware concept](https://nextjs.org/docs/advanced-features/middleware) and I suspect we might want to create custom hooks in the long run.